facade/src/test/java/org/jboss/pnc/facade/providers/SCMRepositoryProviderTest.java

```diff
@@ -0,0 +1,292 @@
+                "git+ssh://" + UUID.randomUUID().toString() + ".eu",
+                true));
+
+        fillRepository(list);
+    }
+
+    @Override
+    protected AbstractProvider provider() {
+        return provider;
+    }
+
+    @Override
+    protected Repository<RepositoryConfiguration, Integer> repository() {
+        return repository;
+    }
+
+    @Test
+    public void testStoreNewRepositoryWithoutId() {
+        // when
+        SCMRepository toCreate = createNewSCMRepository(
+                "https://" + UUID.randomUUID().toString() +".ca",
+                "git+ssh://" + UUID.randomUUID().toString() + ".eu",
+                true,
+                null);
+
+        SCMRepository created = provider.store(toCreate);
+
+        // then
+        assertThat(created).isNotNull();
+        assertThat(created.getId()).isNotNull().isNotEmpty();
+        assertThat(created.getInternalUrl()).isEqualTo(toCreate.getInternalUrl());
+        assertThat(created.getExternalUrl()).isEqualTo(toCreate.getExternalUrl());
+        assertThat(created.getPreBuildSyncEnabled()).isEqualTo(toCreate.getPreBuildSyncEnabled());
+    }
+
+    @Test
+    public void testStoreNewRepositoryWithIdShouldFail() {
+
+        // when
+        SCMRepository toCreate = createNewSCMRepository(
+                "https://" + UUID.randomUUID().toString() +".ca",
+                "git+ssh://" + UUID.randomUUID().toString() + ".eu",
+                true,
+                Integer.toString(entityId++));
+
+        // then
+        assertThatThrownBy(() -> provider.store(toCreate))
+                .isInstanceOf(InvalidEntityException.class);
+    }
+
+    @Test
+    public void testStoreNewRepositoryWithInvalidInternalUrlShouldFail() {
+
+        // when
+        SCMRepository toCreate = createNewSCMRepository(
+                "https://" + UUID.randomUUID().toString() +".ca",
+                "noway" + UUID.randomUUID().toString(),
+                true,
+                null);
+
+        // then
+        assertThatThrownBy(() -> provider.store(toCreate))
+                .isInstanceOf(InvalidEntityException.class);
+    }
+
+    @Test
+    public void testUpdate() {
+
+        // when
+        boolean newPreSync = ! mock.isPreBuildSyncEnabled();
+        SCMRepository toUpdate =
+                createNewSCMRepository(mock.getExternalUrl(), mock.getInternalUrl(), newPreSync, mock.getId().toString());
+
+        provider.update(toUpdate.getId(), toUpdate);
+        SCMRepository updated = provider.getSpecific(toUpdate.getId());
+
+        // then
+        assertThat(updated.getId()).isEqualTo(mock.getId().toString());
+        assertThat(updated.getExternalUrl()).isEqualTo(mock.getExternalUrl());
+        assertThat(updated.getInternalUrl()).isEqualTo(mock.getInternalUrl());
+        assertThat(updated.getPreBuildSyncEnabled()).isEqualTo(newPreSync);
+    }
+
+    @Test
+    public void testUpdateWithInvalidInternalUrlShouldFail() {
```

Comment:
Update of internal URL is not possible on DB level anyway.

Reply:
Good catch! Changed test to: `testUpdateWithUpdatedInternalUrlShouldFail`